packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1943,47 +1943,43 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
            _textLayoutLastMinWidth == constraints.minWidth,
       'Last width ($_textLayoutLastMinWidth, $_textLayoutLastMaxWidth) not the same as max width constraint (${constraints.minWidth}, ${constraints.maxWidth}).');
     final TextRange word = _textPainter.getWordBoundary(position);
+    final String originalText = textSelectionDelegate.textEditingValue.text;
```

Comment:
You can do something like this
```
final String? effectiveText = text?.text ?? readOnly ? textSelectionDelegate.textEditingValue.text : null;
```

Reply:
also add a little bit explanation on why we grab textSelectionDelegate.textEditingValue.text if it is read only

Reply:
Although i am still confused at why the text?.text would be different  from textSelectionDelegate.textEditingValue.text for selectable text

Reply:
ok so i figure out why text?.text is empty because selectable text uses textspan with children.

instead of grabbing the value from text?.text, it should grab value from text?.toPlainText(). if you use that you don't need to check the text editing value

Reply:
Wow that's awesome! Thanks for finding that. I just tried it out and it works well.